src/Atmos/Model/multiphysics_types.jl

```diff
@@ -19,3 +22,100 @@ struct PressureGradient{PV <: Momentum} <: TendencyDef{Flux{FirstOrder}, PV} end
 struct Pressure{PV <: Energy} <: TendencyDef{Flux{FirstOrder}, PV} end
 
 struct Advect{PV} <: TendencyDef{Flux{FirstOrder}, PV} end
+
+
+export RemovePrecipitation
+"""
+    RemovePrecipitation{PV} <: TendencyDef{Source, PV}
+A sink to `q_tot` when cloud condensate is exceeding a threshold.
+The threshold is defined either in terms of condensate or supersaturation.
+The removal rate is implemented as a relaxation term
+in the Microphysics_0M module.
+The default thresholds and timescale are defined in CLIMAParameters.jl.
+"""
+struct RemovePrecipitation{PV <: Union{Mass, Energy, TotalMoisture}} <:
+       TendencyDef{Source, PV}
+    " Set to true if using qc based threshold"
+    use_qc_thr::Bool
+end
+
+RemovePrecipitation(b::Bool) = (
+    RemovePrecipitation{Mass}(b),
+    RemovePrecipitation{Energy}(b),
+    RemovePrecipitation{TotalMoisture}(b),
+)
```

Comment:
I think this is nice. It clearly shows what state variables are affected by each source term.